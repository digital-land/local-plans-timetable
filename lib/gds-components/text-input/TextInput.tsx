/**
 * GDS component: https://design-system.service.gov.uk/components/text-input/
 */

interface TextInputProps {
  label: string;
  onChange: (value: string) => void;
  value: string;
  hint?: string;
  showLabel?: boolean;
}

export const TextInput = ({
  label,
  onChange,
  value,
  hint,
  showLabel = true,
}: TextInputProps) => (
  <div className="govuk-form-group">
    <label className="govuk-label">
      {showLabel && label}
      <input
        className="govuk-input"
        type="text"
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
    </label>
    {hint && (
      <div id="more-detail-hint" className="govuk-hint">
        {hint}
      </div>
    )}
  </div>
);
