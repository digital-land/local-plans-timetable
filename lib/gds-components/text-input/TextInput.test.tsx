import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { TextInput } from "./TextInput";

it("returns the plain text value", async () => {
  const onChange = jest.fn();
  const { asFragment } = render(
    <TextInput label="input" onChange={onChange} value="" />
  );

  await userEvent.type(screen.getByLabelText("input"), "a");

  expect(asFragment()).toMatchSnapshot();
  expect(onChange).toHaveBeenCalledWith("a");
});
