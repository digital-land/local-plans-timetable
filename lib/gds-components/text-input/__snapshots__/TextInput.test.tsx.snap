// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns the plain text value 1`] = `
<DocumentFragment>
  <div
    class="govuk-form-group"
  >
    <label
      class="govuk-label"
      for="text-input"
    >
      input
      <input
        class="govuk-input"
        id="text-input"
        type="text"
        value=""
      />
    </label>
  </div>
</DocumentFragment>
`;
